src/main/java/io/kestra/plugin/hightouch/connections/Sync.java

```diff
@@ -0,0 +1,213 @@
+package io.kestra.plugin.hightouch.connections;
+
+import io.kestra.core.models.annotations.Example;
+import io.kestra.core.models.annotations.Plugin;
+import io.kestra.core.models.annotations.PluginProperty;
+import io.kestra.core.models.executions.metrics.Counter;
+import io.kestra.core.models.tasks.RunnableTask;
+import io.kestra.core.runners.RunContext;
+import io.kestra.core.utils.Await;
+import io.kestra.plugin.hightouch.AbstractHightouchConnection;
+import io.kestra.plugin.hightouch.models.Run;
+import io.kestra.plugin.hightouch.models.RunDetails;
+import io.kestra.plugin.hightouch.models.RunDetailsResponse;
+import io.kestra.plugin.hightouch.models.RunStatus;
+import io.kestra.plugin.hightouch.models.SyncDetailsResponse;
+
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.uri.UriTemplate;
+import io.swagger.v3.oas.annotations.media.Schema;
+import lombok.*;
+import lombok.experimental.SuperBuilder;
+import org.apache.commons.lang3.time.DurationFormatUtils;
+import org.slf4j.Logger;
+
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static io.kestra.core.utils.Rethrow.throwSupplier;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Schema(
+    title = "Trigger HighTouch sync"
+)
+@Plugin(
+    examples = {
+        @Example(
+            code = {
+                "token: YOUR_API_TOKEN",
+                "syncId: 1127166",
+            }
+        )
+    }
+)
+public class Sync extends AbstractHightouchConnection implements RunnableTask<Sync.Output> {
+    private static final List<RunStatus> ENDED_STATUS = List.of(
+        RunStatus.FAILED,
+        RunStatus.CANCELLED,
+        RunStatus.SUCCESS
+    );
+
+    @Schema(
+        title = "The sync id to trigger run"
+    )
+    @PluginProperty(dynamic = true)
+    private Long syncId;
```

Comment:
It seems it is a mandatory property, if so please add `@NotNull` from javax.constraints